### PR TITLE
Terminate users with expired or revoked certificates. Ticket STRDEV-516

### PR DESCRIPTION
**Pull request by fernando-guerrero-leon**
These are new scripts that check the client certificates and verify whether or not they are expired and/or revoked, in any of those cases the user session is terminated.

### CODE REVIEW

File: to_copy/usr/bin/boot-expired-revoked-certs.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+terminate_expired_certs() {
+    cat $SESSION_MAP | while read line; do
+        end_date=$( echo $line | cut -d ',' -f 5 )
+        if [ -n "$end_date" ]; then
+            now_epoch=$( date +%s )
+            end_date_epoch=$( date -d "$end_date" +%s )
+            if [[ $now_epoch > $end_date_epoch ]]; then
+                session=$( echo $line | cut -d ',' -f 1 )
+                kill_session $session
```
Comment (unixeO) [inline to_copy/usr/bin/boot-expired-revoked-certs.sh:46]: You have a missing closing fi.

File: to_copy/usr/bin/obtain-cert-end-date.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ((count++))
+    [ $count -gt 5 ] && fail "Could not get a lock"
+done
+echo "$$" > $LOCKFILE
+
+# Verify to only use the end-entity certificate in the chain, discarding CA
+# certificates
+[ "$depth" -eq 0 ] && {
+
+    end_date=$(openssl x509 -noout -in "$peer_cert" -enddate | cut -d "=" -f 2)
```
Comment (unixeO) [inline to_copy/usr/bin/obtain-cert-end-date.sh:32]: $peer_cert it's used before defined?

File: to_copy/usr/bin/openvpn-on-connect.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+done
+echo "$$" > $LOCKFILE
+
+# Prune old records with the same common name
+grep -v $common_name $SESSION_MAP > tmp
+mv -f tmp $SESSION_MAP
+
+serial_0=$(echo $tls_serial_hex_0 |tr -d : | tr '[:lower:]' '[:upper:]')
+serial_1=$(echo $tls_serial_hex_1 |tr -d : | tr '[:lower:]' '[:upper:]')
+serial_2=$(echo $tls_serial_hex_2 |tr -d : | tr '[:lower:]' '[:upper:]')
```
Comment (unixeO) [inline to_copy/usr/bin/openvpn-on-connect.sh:36]: $common_name,  $tls_serial_hex_0, $tls_serial_hex_1, $tls_serial_hex_2, looks like unresolved variables?

File: to_copy/usr/bin/boot-expired-revoked-certs.sh (hunk 1 above)
Comment (fernando-guerrero-leon) [reply to_copy/usr/bin/boot-expired-revoked-certs.sh:46]: Done

File: to_copy/usr/bin/obtain-cert-end-date.sh (hunk 2 above)
Comment (fernando-guerrero-leon) [reply to_copy/usr/bin/obtain-cert-end-date.sh:32]: This is a environmental variable set by OpenVPN

File: to_copy/usr/bin/openvpn-on-connect.sh (hunk 3 above)
Comment (fernando-guerrero-leon) [reply to_copy/usr/bin/openvpn-on-connect.sh:36]: As in the previous comment, these are also environmental variables set by OpenVPN

**Comment by fernando-guerrero-leon** [comment]
I think this PR was not merged and closed by mistake

**Pull request closed without merging**